### ISSUE TITLE
Show if RPi has pending Reporter Daemon Update

### ISSUE THREAD

**Issue by ironsheep** [ironsheep/lovelace-rpi-monitor-card#36]
**Is your feature request related to a problem? Please describe.**

While the card shows the status of the RPi it does not let us know if the Daemon script reporting these details is up-to-date

**Describe the solution you'd like**

The Daemon reports its current version. In the card, compare the reported version to the available version. Show update to v# (when a new version is available)

**Describe alternatives you've considered**

The user can always log into the RPi to tell. or can look at the Daemon repo to learn the current version than can click on each card to tell if up to date, but this is a lot of work.  Alternatively, the user can just update all RPIs periodically.  This indicator is the best solution, tho'

**Additional context**

Nothing more...

**Comment by ironsheep** [comment]
An initial implementation of this feature is now committed and appears in v1.2.9 and later.

### I'm looking for feedback. 

Here is what the indicator looks like now:

![Screen Shot 2023-02-08 at 02 41 03](https://user-images.githubusercontent.com/540005/217493037-62176236-7350-4212-9810-6bff7e30d02c.png)

(The RED text)  Does this format read ok? Does it make sense?

The indicator, shown, says: `v1.6.0 - (v1.6.2 avail.)`
The indicator is only shown when an update is pending. it will be blank when the RPi daemon is at the latest version.

**Comment by bsimmo** [comment]
OK so updated to test, no the layout doesn't work.
This is what it looks like for me
Also as a side the dates etc are not following locality on the card, I'm in the UK and it seems to have the US date setup. 

![Screenshot_20230208_215718](https://user-images.githubusercontent.com/11537335/217660587-3b2d9d75-01e7-40a3-b559-1d42020635f7.jpg)

**Comment by ironsheep** [comment]
@bsimmo thank you for your thoughts on this.

I'm looking for solutions. My motivation for adding it to these cards and no separate ones is that I manage a farm of 20+ RPi's.  Drilling down to check this on each one is not practical. Seeing at a glance is really guiding of my efforts when I need to know which are not up to date.

If my content was reactive to card width would that help?

**Comment by ironsheep** [comment]
I released a version that should make the Daemon update work as you all expect. If you update the daemon to the latest the flag saying the update needed should go away.

I'm still working on the two underlying issues causing the card update slowdowns and stalls.

More when I make progress on them.

**Comment by bsimmo** [comment]
No idea I'll have a look.
Be all and end all it is for your use.
I have three using this at the moment, but will have more as I get around to it.
Having it as an entity means it can be triggered in or even talked to (whatever the current new term is in HA).
The triggering being the automation so it can come up in a flag somewhere. 
Or
You can have one card that list all the cards and their update status (like I do with IP address and another for WiFi signal)
The future want would be a have Controls so you could click a few button and have the Pi update itself (apt, script etc.) but that would need the Pi to subscribe to the topics from HA. *
OK I have no idea how to fully do this in HA. 
And it maybe going beyond an MQTT script and card. 

*security? No idea but I would keep this disabled by default.

I'll keep testing for you, there must be others using this, hopefully they'll pop in when they see HACs info.

**Comment by wokkeltje13** [comment]
The yellow update available color is verry hard to read, red would be better. 
The script is not updated to version 1.6.2, so my rPI is up to date but the script contains the wrong version, made pull request for the script with the correct version.

**Comment by ironsheep** [comment]
@bsimmo ok, I've updated the layout of the card so that the labels float according to card width. They don't overlap any longer.  I've also changed the yellow for (card update age and Daemon update needed) to orange.

**Comment by brunob45** [comment]
I just installed the reporter (v1.7.2) and the lovelace card (v1.3.1), and the card is reporting an older version to be available.

![image](https://user-images.githubusercontent.com/21108660/218330279-376f1fb1-2501-4d75-85aa-be66ae04af03.png)

**Comment by ironsheep** [comment]
@brunob45 well, huh.  That's unexpected!  
I'll fix this in the next release. I'm releasing more frequently as I'm cleaning up the card code and hunting down a runtime exception that makes our cards misbehave.

Oh, if you have a chance could you capture the output from your Daemon as it's being reported to MQTT?   This is easy to do with something like  **MQTT Explorer** from http://mqtt-explorer.com/

Thanks for reporting this. More soon!

**Comment by brunob45** [comment]
Hi @ironsheep
I installed MQTT explorer and got this output for the "monitor" topic.
```json
{
  "info": {
    "timestamp": "2023-02-12T17:15:33-05:00",
    "rpi_model": "RPi 3 Model B r1.2",
    "ifaces": "e,w,b",
    "host_name": "raspberrypi",
    "fqdn": "raspberrypi",
    "ux_release": "bullseye",
    "ux_version": "5.15.84-v8+",
    "up_time": "4 min",
    "last_update": "2023-02-12T17:30:52-05:00",
    "fs_total_gb": 32,
    "fs_free_prcnt": 41,
    "networking": {
      "eth0": {
        "mac": "b8:27:eb:0e:ce:24"
      },
      "wlan0": {
        "IP": "10.0.0.167",
        "mac": "b8:27:eb:5b:9b:71"
      }
    },
    "drives": {
      "root": {
        "size_gb": 32,
        "used_prcnt": 41,
        "device": "/dev/root",
        "mount_pt": "/"
      }
    },
    "memory": {
      "size_mb": "909.629",
      "free_mb": "282.027"
    },
    "cpu": {
      "hardware": "BCM2835",
      "model": "",
      "number_cores": 4,
      "bogo_mips": "153.60",
      "serial": "00000000560ece24",
      "load_1min_prcnt": 73.8,
      "load_5min_prcnt": 83,
      "load_15min_prcnt": 39
    },
    "throttle": [
      "throttled = 0x0",
      "Not throttled"
    ],
    "temperature_c": 68.2,
    "temp_gpu_c": 68.2,
    "temp_cpu_c": 67.7,
    "reporter": "ISP-RPi-mqtt-daemon v1.7.2",
    "reporter_releases": "v1.7.2,v1.6.2",
    "report_interval": 5
  }
}
```

It may have something to do with `Latest` vs `Stable` version.

**Comment by ironsheep** [comment]
@brunob45 thank you! That tells me it's logic within the card that messed up.  
I appreciate your quick response.   Hopefully, I'll have an update later tomorrow (the 13th)

**Comment by ironsheep** [comment]
Ok, fixes committed and appear in v1.3.2 and later.
